trading_calendars/trading_calendar.py

```diff
@@ -850,24 +973,26 @@ def minute_to_session_label(self, dt, direction="next"):
         pd.Timestamp (midnight UTC)
             The label of the containing session.
         """
+        if isinstance(dt, int):
+            dt = pd.Timestamp(dt, unit="ns")
+
         if direction == "next":
-            if self._minute_to_session_label_cache[0] == dt:
+            if self._minute_to_session_label_cache[0] == dt.value:
                 return self._minute_to_session_label_cache[1]
 
-        idx = searchsorted(self.market_closes_nanos, dt)
+        idx = searchsorted(self.market_closes_nanos, dt.value)
         current_or_next_session = self.schedule.index[idx]
 
         if direction == "next":
-            self._minute_to_session_label_cache = (dt, current_or_next_session)
+            self._minute_to_session_label_cache = (
+                dt.value, current_or_next_session
+            )
             return current_or_next_session
         elif direction == "previous":
-            if not is_open(self.market_opens_nanos, self.market_closes_nanos,
-                           dt):
-                # if the exchange is closed, use the previous session
+            if not self.is_open_on_minute(dt, ignore_breaks=True):
```

Comment:
I notice that we convert int->Timestamp above, then we use only `.value` everywhere here and pass the timestamp to `is_open_on_minute`, which also uses only `.value`. Would it make sense to have both these methods do a Timestamp->int conversion at their starts, and then work with ints?